Drv/Ip/TcpClientSocket.cpp

```diff
@@ -0,0 +1,93 @@
+// ======================================================================
+// \title  TcpClientSocket.cpp
+// \author mstarch
+// \brief  cpp file for TcpClientSocket core implementation classes
+//
+// \copyright
+// Copyright 2009-2020, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+
+#include <Drv/Ip/TcpClientSocket.hpp>
+#include <Fw/Logger/Logger.hpp>
+#include <Fw/Types/BasicTypes.hpp>
+
+#ifdef TGT_OS_TYPE_VXWORKS
+    #include <socket.h>
+    #include <inetLib.h>
+    #include <fioLib.h>
+    #include <hostLib.h>
+    #include <ioLib.h>
+    #include <vxWorks.h>
+    #include <sockLib.h>
+    #include <fioLib.h>
+    #include <taskLib.h>
+    #include <sysLib.h>
+    #include <errnoLib.h>
+    #include <string.h>
+#elif defined TGT_OS_TYPE_LINUX || TGT_OS_TYPE_DARWIN
+    #include <sys/socket.h>
+    #include <unistd.h>
+    #include <arpa/inet.h>
+#else
+    #error OS not supported for IP Socket Communications
+#endif
+
+#include <stdio.h>
+#include <string.h>
+
+namespace Drv {
+
+TcpClientSocket::TcpClientSocket() : IpSocket() {}
+
+SocketIpStatus TcpClientSocket::openProtocol(NATIVE_INT_TYPE& fd) {
+    NATIVE_INT_TYPE socketFd = -1;
+    struct sockaddr_in address;
+
+    // Acquire a socket, or return error
+    if ((socketFd = ::socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        return SOCK_FAILED_TO_GET_SOCKET;
+    }
+    // Set up the address port and name
+    address.sin_family = AF_INET;
+    address.sin_port = htons(this->m_port);
+
+    // OS specific settings
+#if defined TGT_OS_TYPE_VXWORKS || TGT_OS_TYPE_DARWIN
+    address.sin_len = static_cast<U8>(sizeof(struct sockaddr_in));
+#endif
+
+    // First IP address to socket sin_addr
+    if (IpSocket::addressToIp4(m_hostname, &(address.sin_addr)) != SOCK_SUCCESS) {
+        ::close(socketFd);
+        return SOCK_INVALID_IP_ADDRESS;
+    };
+
+    // Now apply timeouts
+    if (IpSocket::setupTimeouts(socketFd) != SOCK_SUCCESS) {
+        ::close(socketFd);
+        return SOCK_FAILED_TO_SET_SOCKET_OPTIONS;
+    }
+
+    // TCP requires connect to the socket to allow for communication
+    if (::connect(socketFd, reinterpret_cast<struct sockaddr*>(&address), sizeof(address)) < 0) {
```

Comment:
If you are trying to call the `IpSocket` base class, the calls should be `this->connect(...)`. `::connect` calls the system call in the global namespace. Am I missing something?